internal/app/machined/pkg/controllers/kubespan/manager.go

```diff
@@ -207,6 +208,13 @@ func (ctrl *ManagerController) Run(ctx context.Context, r controller.Runtime, lo
 			continue
 		}
 
+		if wgClient == nil {
```

Comment:
isn;t it always nil?

Reply:
this is inside a `for` loop, so it should be `nil` only the first time it enters the loop, and on subsequent loops it will be already initialized

we're simply deferring initialization to the moment we need `wgCllient` for real: e.g. if KubeSpan is not enabled, no need even to create it

Reply:
the github diff was confusing. Thanks :+1: